src/DependencyInjection/FOSHttpCacheExtension.php

```diff
@@ -398,14 +398,27 @@ private function loadVarnish(ContainerBuilder $container, XmlFileLoader $loader,
     {
         $this->createHttpDispatcherDefinition($container, $config['http'], $this->getAlias().'.proxy_client.varnish.http_dispatcher');
         $options = [
-            'tags_header' => $config['tags_header'],
+            'tag_mode' => $config['tag_mode'],
         ];
+
+        // If a 'tags_header' setting has been provided use that. Otherwise the FOS http-cache library automatically selects the header to use based on the 'tag_mode' setting
+        if (isset($config['tags_header'])) {
```

Comment:
i prefer `array_key_exists` over `isset` because it will notice the problem when `$config` is not defined (e.g. after refactoring). could you change to that?